iothub/service/src/ServiceClient.cs

```diff
@@ -52,6 +59,82 @@ public static ServiceClient CreateFromConnectionString(string connectionString,
             return CreateFromConnectionString(connectionString, TransportType.Amqp, options);
         }
 
+#if !NET451
+
+        /// <summary>
+        /// Creates a <see cref="ServiceClient"/> using AAD token credentials and the specified transport type.
+        /// </summary>
+        /// <param name="hostName">IoT hub host name.</param>
+        /// <param name="credential">AAD <see cref="TokenCredential"/> to authenticate with IoT hub.</param>
+        /// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+        /// <param name="transportSettings">Specifies the AMQP_WS and HTTP proxy settings for service client.</param>
+        /// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+        /// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+        public static ServiceClient Create(
+            string hostName,
+            TokenCredential credential,
+            TransportType transportType,
+            ServiceClientTransportSettings transportSettings = default,
+            ServiceClientOptions options = default)
+        {
+            if (string.IsNullOrEmpty(hostName))
+            {
+                throw new ArgumentNullException($"{nameof(hostName)} is null or empty");
```

Comment:
Prefer this overload: https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=net-5.0#System_ArgumentNullException__ctor_System_String_System_String_

Reply:
The does not take me to a specific overload. This one?
public ArgumentNullException (string paramName, string message);

Reply:
yes